Tests/MapboxMapsTests/Snapshot/SnapshotterTests.swift

```diff
@@ -1,28 +1,127 @@
+            mapSnapshotter: mockMapSnapshotter)
     }
 
     override func tearDown() {
         snapshotter = nil
         mapboxObservableProviderStub = nil
+        mockMapSnapshotter = nil
         super.tearDown()
     }
 
+    // Test snapshot start invokes mockMapSnapshotter startStub
+    func testSnapshotterStartInvocation() {
+        snapshotter.start(overlayHandler: nil) { (_) in
+            XCTAssertNotNil(self.mockMapSnapshotter.startStub.defaultReturnValue)
+        }
+
+        XCTAssertEqual(mockMapSnapshotter.startStub.invocations.count, 1)
+    }
+
+    func testSnapshotterCompletionInvocationFailed() {
+
+        let options = MapSnapshotOptions(size: CGSize.init(width: 300, height: 300), pixelRatio: 2)
+        snapshotter = Snapshotter(options: options, mapboxObservableProvider: mapboxObservableProviderStub.call(with:), mapSnapshotter: mockMapSnapshotter)
+
+        let resultString = "FAILED"
+        mockMapSnapshotter.startStub.defaultSideEffect = { invocation in
+            invocation.parameters(Expected(error: resultString as NSString))
+        }
+
+        snapshotter.start(overlayHandler: nil) { (result) in
+            switch result {
+            case .failure:
+                print("Successful test")
+
+                break
+            case .success:
+                XCTFail()
+            }
+        }
+    }
+
+    func testSnapshotterCancel() {
+        snapshotter.cancel()
+        XCTAssertEqual(mockMapSnapshotter.cancelSnapshotterStub.invocations.count, 1)
+    }
+
+    func testSnapshotterSize() {
+        let size = CGSize(width: 200, height: 200)
+
+        snapshotter.mapSnapshotter.setSizeFor(.init(size))
+
+        mockMapSnapshotter.getSizeStub.defaultReturnValue = Size(size)
+        XCTAssertEqual(snapshotter.snapshotSize, size)
+
+        snapshotter.snapshotSize = size
+        XCTAssertEqual(mockMapSnapshotter.setSizeStub.invocations[0].parameters, Size(size))
+    }
+
+    func testSnapshotterTileMode() {
+
+        snapshotter.mapSnapshotter.setTileModeForSet(true)
+
+        XCTAssertEqual(mockMapSnapshotter.setTileModeStub.invocations.count, 1)
+        XCTAssertEqual(snapshotter.mapSnapshotter.isInTileMode(), mockMapSnapshotter.isInTileMode())
+    }
+
+    func testSnapshotterSetCamera() {
+        let cameraOptions = CameraOptions(center: CLLocationCoordinate2D(latitude: 38, longitude: -76), padding: .zero, anchor: .zero, zoom: 15, bearing: .zero, pitch: 90)
+        let coreCameraOptions = MapboxCoreMaps.CameraOptions(cameraOptions)
+
+        snapshotter.setCamera(to: cameraOptions)
+
+        XCTAssertEqual(mockMapSnapshotter.setCameraStub.invocations.count, 1)
+        XCTAssertEqual(snapshotter.mapSnapshotter.getStyleDefaultCamera(), mockMapSnapshotter.getStyleDefaultCameraStub.defaultReturnValue)
+    }
+
+    //Test snapshot coordinate bounds for camera match those of mock
+    func testSnapshotterCoordinateBoundsForCamera() {
```

Comment:
This test should setup mapSnapshottter to return a particular options for the coordinateBoundsForCamera stub, get coordinateBoundsForCamera from snapshotted and verify that coordinateBoundsForCamera stub of mapSnapshotter was called and the return value of the call to Snapshotter.coordinateBoundsForCamera is the same as coordinateBoundsForCamera stub of mapSnapshotter was setup with.